crates/torii/core/src/processors/register_model.rs

```diff
@@ -31,15 +31,14 @@ impl<P: Provider + Sync + 'static> EventProcessor<P> for RegisterModelProcessor
         let name = parse_cairo_short_string(&event.data[0])?;
         println!("Register {name}");
 
-        // TODO: remove BlockId as argument
-        let model = world.model(&name, BlockId::Tag(BlockTag::Latest)).await?;
-        let schema = model.schema(BlockId::Tag(BlockTag::Latest)).await?;
-        let layout = model.layout(BlockId::Tag(BlockTag::Latest)).await?;
+        world.set_call_block_id(BlockId::Tag(BlockTag::Latest));
```

Comment:
Hm this is an interesting pattern. Why did you decide to go this path vs passing for each call?

Reply:
Adding an extra-parameter to function in a sense makes it less natural to call a function of our contract as we have to know that the `abigen` will add this extra param.

But at the same time, having the `set_call_block_id` pattern is a "limitation"/"constraint" when the instance is shared, as we need a `mut` reference to adjust the block id, and this may cause some races without adding mutex for instance.

So... I would say that it was an experiment, and after making more test I am not sure what can be the best approach. Or maybe `abigen` may give this option to the user?